data-processing/scripts/dsk/convert_dsk_to_jsonl.py

```diff
@@ -0,0 +1,227 @@
+        **kwargs (dict): Additional arguments
+
+    Returns:
+        str: JSONL line with the file content
+    """
+    text = file_path.read_text()
+    text = re.sub(r"(\n\s)+", "\n", text)
+    metadata = build_metadata(file_path)
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
+
+
+def process_document(
+    file: Path,
+    source: str,
+    **kwargs: dict[str, Any],
+) -> str | None:
+    """
+    Process a single file, converting it into a JSONL string.
+
+    Args:
+        file (Path): Path to the input file.
+        source (str): Source identifier for the data.
+        **kwargs (dict): Extra arguments
+
+    Returns:
+        str | None: JSONL string if the file is processed successfully, else None.
+    """
+    doc_converter = kwargs.get("converter", build_document_converter())
+    try:
+        result = doc_converter.convert(file)
+
+        metadata = build_metadata(result.input)
+
+        file_content = ""
+        # Iterate the elements in reading order, including hierarchy level
+        for item, _ in result.document.iterate_items():
+            if isinstance(item, TextItem):
+                if item.text.strip():
+                    file_content += item.text
+            elif isinstance(item, TableItem):
+                table_df: pd.DataFrame = item.export_to_dataframe()
+                dups = find_near_duplicates(table_df, 0.8)
+                column_counts = {}
+                table_df.columns = [
+                    make_unique(col, column_counts) for col in table_df.columns
+                ]
+                columns_to_keep = table_df.columns.copy(deep=True)
+                for pair in dups:
+                    if pair[1] in columns_to_keep:
+                        columns_to_keep = columns_to_keep.drop(pair[1])
+
+                df_cleaned = table_df[list(columns_to_keep)]
+                df_cleaned = remove_newlines(df_cleaned)
+                file_content += df_cleaned.to_markdown(index=False, tablefmt="github")
+            file_content += "\n\n"
+
+        # Create and return JSONL entry
+        return json.dumps(
+            asdict(create_JSONL(file_content, source, metadata)),
+            ensure_ascii=False,
+        )
+    except Exception as e:
+        logger.error(f"Failed to process file {file}: {e}")
+        return None
+
+
+SUFFIX2METHOD = {
+    ".pdf": process_document,
+    ".html": process_html,
+    ".docx": process_document,
+    ".epub": process_epub,
+    ".txt": process_txt,
+    ".pptx": process_document,
+    ".md": process_document,
+}
+
+
+def process_file(file: Path, dsk_client: str, **kwargs: dict) -> str | None:
+    if file.is_dir():
+        return None
+
+    suffix = file.suffix
+    method = SUFFIX2METHOD.get(suffix)
+
+    if not method:
+        logger.warning(f"Unsupported file type: {suffix} - for file: {file!s}")
+        return None
+
+    return method(file, dsk_client, **kwargs)
```

Comment:
The `SUFFIX2METHOD` is only used in `process_file`, so maybe it should be moved in there:

```suggestion
def process_file(file: Path, dsk_client: str, **kwargs: dict) -> str | None:
    if file.is_dir():
        return None

    suffix = file.suffix
    method = {
        ".pdf": process_document,
        ".html": process_html,
        ".docx": process_document,
        ".epub": process_epub,
        ".txt": process_txt,
        ".pptx": process_document,
        ".md": process_document,
    }.get(suffix)

    if not method:
        logger.warning(f"Unsupported file type: {suffix} - for file: {file!s}")
        return None

    return method(file, dsk_client, **kwargs)
```